### ISSUE TITLE
Support for description fields in .saveAsBigQuery function

### ISSUE THREAD

**Issue by ankscircle** [spotify/scio#340]
When saving Dataflow output as a BigQuery table, we right now use case class. It would be great to have support to add description fields for columns in the BigQuery table. So something like .saveAsBigQuery(<table-ref>, <schema>) where schema has the ability to setDescriptions on the table fields.

**Comment by ravwojdyla** [comment]
This is already kinda supported, but not very dev friendly:

```
val writeSchema = <BigQueryType>.schema
writeSchema.getFields().asScala.find(_.getName == "fieldName").map(_.setDescription("my new desc"))

sc.<logic>.saveAsBigQuery(<tbl-ref>, writeSchema)
```

Could we place the description inside the macro - which would make it static? In your case @ankscircle how do you obtain description for the fields - would they be static like:

```
@BigQueryType.toTable case class Foo(foo: Int @description("count of stream"), bar: String @description("user name"))
```

or are the descriptions dynamic/parametrized?

**Comment by nevillelyh** [comment]
Can you use annotations like that?
I don't see a use case of description being parameterized, because that would imply type also being parameterized which is impossible?

**Comment by ravwojdyla** [comment]
@ankscircle ping

**Comment by ankscircle** [comment]
@nevillelyh @ravwojdyla That makes sense. The description need not be parametrize. They won't change. So something like `@description` looks great! And would also help maintain the schema and is more readable! 👍

**Comment by nevillelyh** [comment]
Fixed in #349
